From 72ff679d90861283e3f353d25fd10c3d130ed07b Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 1 Nov 2015 11:45:50 -0500
Subject: [PATCH 0656/1936] PAD/Rtosc: Fix GCC 4.8.x Bug

decltype() was/is buggy when used in lambdas in class static member initialization:
http://gcc.gnu.org/bugzilla/show_bug.cgi?id=65727
---
 src/Params/PADnoteParameters.cpp | 13 ++++++++-----
 src/Params/PADnoteParameters.h   |  4 ++--
 2 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/src/Params/PADnoteParameters.cpp b/src/Params/PADnoteParameters.cpp
index 2a16203d..6e2cf5f1 100644
--- a/src/Params/PADnoteParameters.cpp
+++ b/src/Params/PADnoteParameters.cpp
@@ -35,7 +35,7 @@ using namespace rtosc;
 
 
 #define rObject PADnoteParameters
-const rtosc::Ports PADnoteParameters::realtime_ports =
+static const rtosc::Ports realtime_ports = 
 {
     rRecurp(FreqLfo, "Frequency LFO"),
     rRecurp(AmpLfo,   "Amplitude LFO"),
@@ -119,8 +119,7 @@ const rtosc::Ports PADnoteParameters::realtime_ports =
         }},
 
 };
-
-const rtosc::Ports PADnoteParameters::non_realtime_ports =
+static const rtosc::Ports non_realtime_ports =
 {
     //Harmonic Source Distribution
     rRecurp(oscilgen, "Oscillator"),
@@ -220,10 +219,14 @@ const rtosc::Ports PADnoteParameters::non_realtime_ports =
             delete[] tmp;}},
 };
 
+const rtosc::Ports &PADnoteParameters::non_realtime_ports = ::non_realtime_ports;
+const rtosc::Ports &PADnoteParameters::realtime_ports     = ::realtime_ports;
+
+
 const rtosc::MergePorts PADnoteParameters::ports =
 {
-    &PADnoteParameters::non_realtime_ports,
-    &PADnoteParameters::realtime_ports
+    &non_realtime_ports,
+    &realtime_ports
 };
 
 
diff --git a/src/Params/PADnoteParameters.h b/src/Params/PADnoteParameters.h
index 5cffec12..659e2d59 100644
--- a/src/Params/PADnoteParameters.h
+++ b/src/Params/PADnoteParameters.h
@@ -166,8 +166,8 @@ class PADnoteParameters:public Presets
                              std::function<bool()> do_abort);
 
         static const rtosc::MergePorts ports;
-        static const rtosc::Ports      non_realtime_ports;
-        static const rtosc::Ports      realtime_ports;
+        static const rtosc::Ports     &non_realtime_ports;
+        static const rtosc::Ports     &realtime_ports;
 
     private:
         void generatespectrum_bandwidthMode(float *spectrum,
-- 
2.47.0

